recipes/entityx/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.33.0"
+
+
+class EntityXConan(ConanFile):
+    name = "entityx"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/alecthomas/entityx/"
+    topics = ("entity", "conan")
+    license = "MIT"
+    description = "EntityX is an EC system that uses C++11 features to provide type-safe component management, event delivery, etc."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake"
+
+    exports_sources = "CMakeLists.txt"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("entityx-" + self.version, self._source_subfolder)
+
+    def validate(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Windows shared library do not export all symbols")
```

Comment:
Therefore why using `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` in CMakeLists wrapper?

Reply:
I tried, but it fails with link errors:
```
test_package.obj : error LNK2019: unresolved external symbol "protected: static unsigned __int64 entityx::BaseEvent::family_counter_" (?family_counter_@BaseEvent@entityx@@1 _KA) referenced in function "public: class entityx::Entity __cdecl entityx::EntityManager::create(void)"
```

The fact that the DLL was not ``cmake.install()`` correctly, and I had to add:
```python
self.copy("*entity*.dll", src=self._build_subfolder, dst="bin", keep_path=False)
```
Sounds like an indicator that might not be correctly supported?

Reply:
Related: https://github.com/alecthomas/entityx/issues/225

but doesn't seem to have good alternative in MSVC: https://developercommunity.visualstudio.com/t/implement-zcdllexportinlines-aka-fvisibility-inlin/374754

Reply:
If there are global symbols not manually exported, CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS can't help, so I would remove CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS from CMake wrapper, and keep ConanInvalidConfiguration if msvc and shared.